allennlp/training/trainer.py

```diff
@@ -628,10 +630,16 @@ def _validation_loss(self) -> Tuple[float, int]:
         batches_this_epoch = 0
         val_loss = 0
         for batch in val_generator_tqdm:
-            batches_this_epoch += 1
 
             loss = self._batch_loss(batch, for_training=False)
-            val_loss += loss.data.cpu().numpy()
+            if loss is not None:
+                # You shouldn't necessarily have to compute a loss for validation, so we allow for
```

Comment:
Actually when is this used?

Reply:
This is used by the wikitables parser (which is why it's part of this PR). Our training loss is maximum marginal likelihood, but we don't have a set of candidate logical forms for all of the validation instances. We still can calculate accuracy, though, and we want to include all of the instances in the validation set, to get accurate metrics. So we need the training code to not crash when we're evaluating instances at validation time for which we can't actually compute a loss.

Reply:
makes sense 👍